Perpetuum.DataDumper/DataDumper.cs

```diff
@@ -0,0 +1,604 @@
+using Perpetuum.Groups.Alliances;
+using System.Reflection;
+using NPOI.SS.UserModel;
+
+namespace Perpetuum.DataDumper
+{
+    public partial class DataDumper
+    {
+        private IContainer container;
+        EntityFactory entityFactory;
+        IEntityServices entityServices;
+        IExtensionReader extensionReader;
+        IEntityDefaultReader defaultReader;
+        Dictionary<string, string> itemNames;
+        ISparkRepository sparkRepository;
+
+        // Some static lists for helpers
+        public static SlotFlags[] sizeFlags = new SlotFlags[] { SlotFlags.small, SlotFlags.medium, SlotFlags.large };
+        public static SlotFlags[] typeFlags = new SlotFlags[] { SlotFlags.turret, SlotFlags.missile, SlotFlags.melee, SlotFlags.industrial, SlotFlags.ew_and_engineering };
+        public static SlotFlags[] locationFlags = new SlotFlags[] { SlotFlags.head, SlotFlags.chassis, SlotFlags.leg };
+        public static List<CategoryFlags> ammoWeaponCategories = new List<CategoryFlags> {
+                                                                CategoryFlags.cf_railgun_ammo,
+                                                                CategoryFlags.cf_laser_ammo,
+                                                                CategoryFlags.cf_missile_ammo,
+                                                                CategoryFlags.cf_projectile_ammo };
+
+        private static List<DataDumpConfiguration> views = new List<DataDumpConfiguration> {
+            new DataDumpConfiguration { ViewType = typeof(AmmoWeaponDataView), ViewCategory = "cf_ammo" },
+            new DataDumpConfiguration { ViewType = typeof(ModuleWeaponDataView), ViewCategory = "cf_weapons" }
+            
+        };
+
+        public DataDumper(IContainer container)
+        {
+            this.container = container;
+
+            entityFactory = container.Resolve<EntityFactory>(); 
+
+            extensionReader = container.Resolve<IExtensionReader>();
+
+            defaultReader = container.Resolve<IEntityDefaultReader>();
+
+            var productionDataReader = container.Resolve<IProductionDataAccess>();
+
+            sparkRepository = container.Resolve<ISparkRepository>();
+
+            entityServices = container.Resolve<IEntityServices>();
+
+            // var getEd = defaultReader.GetByName("def_named1_small_armor_repairer");
+            // var getCats = String.Join(";", getEd.CategoryFlags.GetCategoryFlagsTree().Where(x => x.ToString() != "undefined").Select(x => x.ToString()).ToList());
+            // var testMissile = productionDataReader.ProductionComponents[64];
+            // var testRobot = productionDataReader.ProductionComponents[193]; 
+            // var testRobot2 = productionDataReader.ProductionComponents[208];
+
+            itemNames = new Dictionary<string, string>();
+
+            var dataLines = System.IO.File.ReadAllLines(@"C:\Users\derek\Desktop\projects\Perpetuum\dictionary.tsv");
```

Comment:
I'll let some stuff slide for this first pass, but this should be revised ;)

Reply:
Actually I see what this is now, and maybe this is a bit trickier because its effectively an arbitrary file that could exist anywhere, but is needed for this step.
Probably allow the user to plug in the path, and/or check for presence in some default relative path and alert the user the file isnt found.